### PR TITLE
Implement `Display` trait

### PR DESCRIPTION
**Pull request by sgeisler**
- [x] refactor and move `to_string` into `Display` trait
- [x] test `new` function and getters

### CODE REVIEW

**Comment by clarkmoody** [comment]
I like what's going on here so far. The range checks on the way in are a good development.

**Comment by sgeisler** [comment]
Once this PR is merged I will rebase [more-cleanups](https://github.com/sgeisler/rust-bech32/tree/more-cleanups) on master and open another PR. These changes do not directly belong in here but lead to better API consistency (transform `&Vec<_>` to `&[_]`) and cleaner code (by applying some clippy hints).

**Comment by apoelstra** [comment]
We can now eliminate `type EncodeResult = Result<String, Error>;` in src/lib.rs.

I would also like a `len()` function that would give me the byte-length in base 256 of the decoded data, to distinguish between scripthash witness programs and pubkeyhash witness programs, but that does not seem appropriate for this PR.

Otherwise ACK.

File: src/lib.rs [hunk 1]
```diff
@@ -297,6 +321,28 @@ mod tests {
...
+        assert_eq!(Bech32::new("".into(), vec![1, 2, 3, 4]), Err(Error::InvalidLength));
+        assert_eq!(Bech32::new("test".into(), vec![30, 31, 35, 20]), Err(Error::InvalidData(35)));
+
+        let both = Bech32::new("".into(), vec![30, 31, 35, 20]);
+        assert!(both == Err(Error::InvalidLength) || both == Err(Error::InvalidData(35)));
+    }
+
+    #[test]
+    fn getters() {
+        let bech: Bech32 = "BC1SW50QA3JX3S".parse().unwrap();
```
Comment (clarkmoody) [inline src/lib.rs]: Very smooth ergonomics here.

File: src/lib.rs [hunk 2]
```diff
@@ -51,37 +48,64 @@
...
+    pub fn hrp(&self) -> &str {
+        &self.hrp
+    }
+
+    /// Returns the data part as `[u8]` but only using 5 bits per byte
+    pub fn data(&self) -> &[u8] {
+        &self.data
+    }
+
+    /// Destructures the `Bech32` struct into it's parts
```
Comment (clarkmoody) [inline src/lib.rs]: Grammar nit: `it's` -> `its`

File: src/lib.rs [hunk 3]
```diff
@@ -51,37 +48,64 @@
...
 pub struct Bech32 {
     /// Human-readable part
-    pub hrp: String,
+    hrp: String,
     /// Data payload
-    pub data: Vec<u8>
+    data: Vec<u8>
 }
 
 type EncodeResult = Result<String, Error>;
```
Comment (clarkmoody) [inline src/lib.rs]: As @apoelstra mentioned, we can eliminate this type.

**Pull request merged**